cg/server/admin.py

```diff
@@ -428,6 +430,86 @@ def view_sample_link(unused1, unused2, model, unused3):
             else ""
         )
 
+    @action(
+        "cancel_samples",
+        "Cancel samples",
+        "Are you sure you want to cancel the selected samples?",
+    )
+    def cancel_samples(self, entry_ids: List[str]):
+        self.write_cancel_comment(entry_ids)
+        families = self.remove_family_samples(entry_ids)
+        self.handle_families_with_multiple_samples(families)
+
+    def handle_families_with_multiple_samples(self, families: List[Family]):
+        family_ids = [family.internal_id for family in families]
+
+        flash(
+            ngettext(
+                f"The cancelled sample was present in families containing other samples",
+                f"Family names: {family_ids}",
+                len(family_ids),
+            )
+        )
+
+    def write_cancel_comment(self, entry_ids: List[str]):
+        try:
+            username = db.user(session.get("user_email")).name
+            date = datetime.now().strftime("%Y-%m-%d")
+            comment = f"Cancelled {date} by {username}"
+
+            query: Query = db.Sample.query.filter(db.Sample.id.in_(entry_ids))
+            sample: Sample
+            for sample in query.all():
+                sample.comment = comment
+            db.commit()
+
+            flash(
+                ngettext(
+                    f"Samples were cancelled.",
+                    f"{len(entry_ids)} samples were cancelled.",
+                    len(entry_ids),
+                )
+            )
+        except Exception as ex:
+            if not self.handle_view_exception(ex):
+                raise
+            flash(gettext(f"Failed to set sample action. {str(ex)}"))
+
+    def remove_families(self, entry_ids: List[str]):
+        query: Query = db.Sample.query.filter(db.Sample.id.in_(entry_ids))
+        families_with_multiple_samples = []
+
+        family: Family
+        for family in query.all():
+            if self.family_contains_one_sample(family):
```

Comment:
If the function returns a bool - use `is`, `was` or `has` as the first word in the function name. It will convey the purpose of the function.
```suggestion
            if self.is_single_sample(case=case):
```